src/ui/articleview.hh

```diff
@@ -35,8 +36,7 @@ class ArticleView: public QWidget
 
   ArticleNetworkAccessManager & articleNetMgr;
   AudioPlayerPtr const & audioPlayer;
-  std::vector< sptr< Dictionary::Class > > const & allDictionaries;
-  Instances::Groups const & groups;
+  DictionaryGroup dictionaryGroup;
```

Comment:
`std::unique_ptr / QScopedPointer` seems better than "`QObject` with `this` as parent" in here. They manage the lifetime equally, but the former don't have bloats.

Reply:
two different ways to the same goal.

Reply:
For objects with no complex deleter, the cost of unique_ptr is zero.